include/pcl_conversions/pcl_conversions.h

```diff
@@ -79,13 +79,13 @@ namespace pcl_conversions {
   inline
   void fromPCL(const pcl::uint64_t &pcl_stamp, ros::Time &stamp)
   {
-    stamp.fromNSec(pcl_stamp * 1e3);  // Convert from us to ns
+    stamp.fromNSec(pcl_stamp * 1000ull);  // Convert from us to ns
   }
 
   inline
   void toPCL(const ros::Time &stamp, pcl::uint64_t &pcl_stamp)
   {
-    pcl_stamp = stamp.toNSec() / 1e3;  // Convert from ns to us
+    pcl_stamp = stamp.toNSec() / 1000ull;  // Convert from ns to us
```

Comment:
1e3 is a double. That creates weird rounding errors. The test case that I added shows the problem. With 1e3 the test fails, with 1000ull it passes.